packages/legacy/core/App/container-api.ts

```diff
@@ -141,7 +136,7 @@ export type TokenMapping = {
   [TOKENS.LOAD_STATE]: LoadStateFn
   [TOKENS.COMP_BUTTON]: Button
   [TOKENS.NOTIFICATIONS]: {
-    useNotifications: () => Array<BasicMessageRecord | CredentialRecord | ProofExchangeRecord | CustomNotification>
+    useNotifications: ({ openIDUri } : NotificationsInputProps) => NotificationReturnType
```

Comment:
I'm not sure this is the best approach with the way `openIDUri` is being passed around through so many components, and used in the notifications hook only sometimes. The hook should get all active notifications not just the ones for a certain connection uri.

Reply:
This is not passing the parameter all over the place, this is defining that this function accepts a parameter of this name and type. I think this is normal to have a function with parameters but if you have another better suggestion, please let me know. 
Considering that OpenID is totally different than DidComm, you cannot use a hooks to listen to credential state and trigger a notification. Its one way flow, with that in mind, to support OpenID in the wallet, we either have to clone every component starting from scanning the QR code all the way to the UI of accepting or verifying the credentials. I avoided that because I didn't want to crowd the wallet with duplicated components and I kept it only for the UI that uses OCA which have to be cloned